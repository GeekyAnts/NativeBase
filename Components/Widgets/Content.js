/* @flow */
'use strict';

import React from 'react';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import NativeBaseComponent from '../Base/NativeBaseComponent';
import computeProps from '../../Utils/computeProps';


export default class Content extends NativeBaseComponent {

  propTypes: {
        padder : React.PropTypes.bool,
    style : React.PropTypes.object
    }

  prepareRootProps() {

    var type = {
      backgroundColor: 'transparent',
      flex: 1
    }

    var defaultProps = {
      style: type
    }

    return computeProps(this.props, defaultProps);
  }

  render() {
    const contentContainerStyle = this.props.contentContainerStyle || {};
    contentContainerStyle.padding = (this.props.padder) ? this.getTheme().contentPadding : 0;
    return(
      <KeyboardAwareScrollView ref={c => this._scrollview = c} resetScrollToCoords={{x:0,y:0}} {...this.prepareRootProps()} contentContainerStyle={contentContainerStyle}>{this.props.children}</KeyboardAwareScrollView>
    );
  }
}
