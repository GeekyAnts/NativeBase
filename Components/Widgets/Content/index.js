/* @flow */
'use strict';

import React from 'react';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import NativeBaseComponent from '../../Base/NativeBaseComponent';
import computeProps from '../../../Utils/computeProps';


export default class Content extends NativeBaseComponent {

  propTypes: {
      padder: React.PropTypes.bool,
      style : React.PropTypes.object,
      disableKBDismissScroll: React.PropTypes.bool
    }

  prepareRootProps() {

    var type = {
      backgroundColor: 'transparent',
      flex: 1
    }

    var defaultProps = {
      style: type,
      resetScrollToCoords: (this.props.disableKBDismissScroll) ? null : {
        x: 0,
        y: 0
      }
    }

    return computeProps(this.props, defaultProps);
  }

  render() {
    const contentContainerStyle = this.props.contentContainerStyle || {};
    contentContainerStyle.padding = (this.props.padder) ? this.getTheme().contentPadding : 0;
    return(
      <KeyboardAwareScrollView automaticallyAdjustContentInsets={false} ref={(c) => {this._scrollview = c; this._root = c;}} {...this.prepareRootProps()} contentContainerStyle={contentContainerStyle}>{this.props.children}</KeyboardAwareScrollView>
    );
  }
}
