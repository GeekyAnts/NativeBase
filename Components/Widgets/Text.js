/* @flow */
'use strict';

import React from 'react';
import {Text} from 'react-native';
import NativeBaseComponent from '../Base/NativeBaseComponent';
import computeProps from '../../Utils/computeProps';


export default class TextNB extends NativeBaseComponent {

	prepareRootProps() {

		var type = { 
			color: this.getTheme().textColor,
			fontSize: this.getTheme().fontSizeBase,
			lineHeight: this.getTheme().lineHeight
		}

		var defaultProps = {
			style: type
		}	  

		return computeProps(this.props, defaultProps);

	}
	render() {
		return(
			<Text {...this.prepareRootProps()}>{this.props.children}</Text>
		);
	}    

}

