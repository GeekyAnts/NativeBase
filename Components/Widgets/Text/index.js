/* @flow */
'use strict';

import React from 'react';
import {Text} from 'react-native';
import NativeBaseComponent from '../../Base/NativeBaseComponent';
import computeProps from '../../../Utils/computeProps';


export default class TextNB extends NativeBaseComponent {

  propTypes: {
        style : React.PropTypes.object
    }

  prepareRootProps() {

    var type = {
      color: this.getContextForegroundColor(),
      fontSize: this.getTheme().fontSizeBase,
      lineHeight: this.getTheme().lineHeight,
      fontFamily: this.getTheme().fontFamily,
      backgroundColor: this.getTheme().textBgColor
    }

    var defaultProps = {
      style: type
    }

    return computeProps(this.props, defaultProps);

  }
  render() {
    return(
      <Text ref={c => this._root = c} {...this.prepareRootProps()}>{this.props.children}</Text>
    );
  }

}
