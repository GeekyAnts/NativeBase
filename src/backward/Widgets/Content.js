import React, { Component } from 'react';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { connectStyle } from '@shoutem/theme';
import mapPropsToStyleNames from '../../Utils/mapPropsToStyleNames';
import NativeBaseComponent from '../Base/NativeBaseComponent';

class Content extends NativeBaseComponent {
  render() {
    return (
      <KeyboardAwareScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={this.props.contentContainerStyle}
        resetScrollToCoords={(this.props.disableKBDismissScroll) ? null : { x: 0, y: 0 }}
        ref={(c) => { this._scrollview = c; this._root = c; }}
        {...this.props}
      >
        {this.props.children}
      </KeyboardAwareScrollView>
    );
  }
}

Content.propTypes = {
  ...KeyboardAwareScrollView.propTypes,
  style: React.PropTypes.object,
  padder: React.PropTypes.bool,
  disableKBDismissScroll: React.PropTypes.bool,
};

const StyledContent = connectStyle('NativeBase.Content', {}, mapPropsToStyleNames)(Content);

export {
  StyledContent as Content,
};
