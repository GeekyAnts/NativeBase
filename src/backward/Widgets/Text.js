/* @flow */


import React from 'react';
import { Text as TextRN } from 'react-native';
import { connectStyle } from '@shoutem/theme';
import mapPropsToStyleNames from '../../Utils/mapPropsToStyleNames';
import NativeBaseComponent from '../Base/NativeBaseComponent';


class Text extends NativeBaseComponent {

  render() {
    return (
      <TextRN ref={c => this._root = c} {...this.props}>{this.props.children}</TextRN>
    );
  }

}

Text.propTypes = {
  ...TextRN.propTypes,
  style: React.PropTypes.object,
};

const StyledText = connectStyle('NativeBase.Text', {}, mapPropsToStyleNames)(Text);

export {
  StyledText as Text,
};
