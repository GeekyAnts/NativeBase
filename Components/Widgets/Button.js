/* @flow */
'use strict';

import React from 'react';
import { TouchableOpacity, Platform } from 'react-native';
import NativeBaseComponent from '../Base/NativeBaseComponent';
import computeProps from '../../Utils/computeProps';
import IconNB from './Icon';
import Icon from './Icon';
import Text from './Text';
import _ from 'lodash';


export default class Button extends NativeBaseComponent {

    propTypes: {
        style : React.PropTypes.object,
        textStyle : React.PropTypes.object,
        block : React.PropTypes.bool,
        primary : React.PropTypes.bool,
        transparent : React.PropTypes.bool,
        success : React.PropTypes.bool,
        danger : React.PropTypes.bool,
        warning : React.PropTypes.bool,
        info : React.PropTypes.bool,
        bordered : React.PropTypes.bool,
        disabled : React.PropTypes.bool,
        rounded : React.PropTypes.bool,
        large : React.PropTypes.bool,
        small : React.PropTypes.bool,
        inputButton : React.PropTypes.bool
    }

    getInitialStyle() {
        return {
            button: {
                paddingVertical: (this.props.vertical) ? 0 :this.getTheme().buttonPadding,
                paddingHorizontal: (this.props.vertical) ? 5 :this.getTheme().buttonPadding+2,
                justifyContent: (this.props.block) ? 'center' : 'space-around',
                flexDirection:  (this.props.vertical) ? 'column' : 'row',
                alignSelf: 'center',
                alignItems: 'center',
                backgroundColor: this.getTheme().btnPrimaryBg,
                elevation: (this.props.transparent || this.props.bordered) ? 0 : 3,
                shadowColor: (this.props.transparent || this.props.bordered) ? undefined : '#000',
                shadowOffset: (this.props.transparent || this.props.bordered) ? undefined : {width: 0, height: 2},
                shadowOpacity: (this.props.transparent || this.props.bordered) ? undefined : 0.2,
                shadowRadius: (this.props.transparent || this.props.bordered) ? undefined : 2
            }
        }
    }

    prepareRootProps() {

        var type = {
            backgroundColor:    (this.props.primary) ? this.getTheme().btnPrimaryBg :
                                ((this.props.transparent) || (this.props.bordered)) ? 'rgba(0,0,0,0)' :
                                (this.props.success) ? this.getTheme().btnSuccessBg :
                                (this.props.danger) ? this.getTheme().btnDangerBg :
                                (this.props.warning) ? this.getTheme().btnWarningBg :
                                (this.props.info) ? this.getTheme().btnInfoBg :
                                (this.props.backgroundColor) ? this.props.backgroundColor :
                                (this.props.disabled) ? this.getTheme().btnDisabledBg :
                                this.getInitialStyle().button.backgroundColor,
            borderRadius: (this.props.rounded) ? this.getTheme().borderRadiusLarge : this.getTheme().borderRadiusBase,
            borderWidth: (this.props.bordered) ? 1 : 0,
            borderColor:    (this.props.primary) ? this.getTheme().primary :
                            (this.props.success) ? this.getTheme().btnSuccessBg :
                            (this.props.danger) ? this.getTheme().btnDangerBg :
                            (this.props.warning) ? this.getTheme().btnWarningBg :
                            (this.props.info) ? this.getTheme().btnInfoBg :
                            this.getInitialStyle().button.backgroundColor,
            height: (this.props.large) ? 60 : (this.props.small) ? 35 : 38,
            alignSelf: (this.props.block) ? 'stretch' : 'flex-start'
        }

        var addedProps = _.merge(this.getInitialStyle().button,type);

        var defaultProps = {
            style: addedProps,
            capitalize: true
        }

        return computeProps(this.props, defaultProps);
    }

    getTextStyle() {

        var mergedStyle = {};
        var btnType = {
            fontFamily: this.getTheme().btnFontFamily,
            marginLeft: (this.props.vertical) ? 0 : (this.iconPresent() && !this.props.iconRight) ? this.getTheme().iconMargin : 0,
            marginRight: (this.iconPresent() && this.props.iconRight) ? this.getTheme().iconMargin : 0,
            color:
                    ((this.props.bordered) && (this.props.primary)) ? this.getTheme().btnPrimaryBg :
                    ((this.props.bordered) && (this.props.success)) ? this.getTheme().btnSuccessBg :
                    ((this.props.bordered) && (this.props.danger)) ? this.getTheme().btnDangerBg :
                    ((this.props.bordered) && (this.props.warning)) ? this.getTheme().btnWarningBg :
                    ((this.props.bordered) && (this.props.info)) ? this.getTheme().btnInfoBg :
                    ((this.props.bordered)) ? this.getTheme().btnPrimaryBg :
                    (this.props.color) ? this.props.color :
                    (this.props.transparent) ? this.getContextForegroundColor() :
                    this.getTheme().inverseTextColor,

            fontSize: (this.props.large) ? this.getTheme().btnTextSizeLarge : (this.props.small) ? this.getTheme().btnTextSizeSmall : this.getTheme().btnTextSize,

            lineHeight: (this.props.large) ? 29 : (this.props.small) ? 16 : this.getTheme().btnLineHeight
        }

        return _.merge(mergedStyle, btnType, this.props.textStyle);
    }

    getIconProps(icon) {

        var defaultStyle = {
            color:
                ((this.props.bordered) && (this.props.primary)) ? this.getTheme().btnPrimaryBg :
                ((this.props.bordered) && (this.props.success)) ? this.getTheme().btnSuccessBg :
                ((this.props.bordered) && (this.props.danger)) ? this.getTheme().btnDangerBg :
                ((this.props.bordered) && (this.props.warning)) ? this.getTheme().btnWarningBg :
                ((this.props.bordered) && (this.props.info)) ? this.getTheme().btnInfoBg :
                (this.props.bordered) ? this.getTheme().btnPrimaryBg :
                (this.props.color) ? this.props.color :
                (this.props.header) ? this.getTheme().toolbarTextColor :
                (this.props.transparent) ? this.getContextForegroundColor() :
                this.getTheme().inverseTextColor,

            fontSize: (this.props.large) ? this.getTheme().iconSizeLarge : (this.props.small) ? this.getTheme().iconSizeSmall : (this.props.inputButton) ? this.getTheme().toolbarIconSize : this.getTheme().iconFontSize-5,
            lineHeight: (this.props.large) ? 52: (this.props.small || this.props.inputButton) ? 22 : this.getTheme().iconLineHeight-9
        }

        var defaultProps = {
            style: defaultStyle
        }

        return computeProps(icon.props, defaultProps);
    }
    iconPresent() {
        var iconComponentPresent = false;
        React.Children.forEach(this.props.children, function (child) {
            if(child.type == Icon)
                iconComponentPresent = true;
        })
        return iconComponentPresent;
    }
    renderChildren() {
        
        if(typeof this.props.children == 'string') {
            return <Text style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? this.props.children : this.props.children.toUpperCase()}</Text>
        }

        else if(this.props.children.type == IconNB) {
            return React.cloneElement(this.props.children, this.getIconProps(this.props.children));
        }

        else if(Array.isArray(this.props.children)) {
            var newChildren = [];

            var childrenArray = React.Children.toArray(this.props.children);

            var iconElement = [];
            iconElement = _.remove(childrenArray, function(item) {
                if(item.type == IconNB) {
                    return true;
                }
            });

            if(this.props.iconRight) {
                if (childrenArray[0].type==undefined) {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
                } else {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
                }

                newChildren.push(<Text key='icon'>{React.cloneElement(iconElement[0], this.getIconProps(iconElement[0]))}</Text>);
            }

            else if(this.props.iconLeft || iconElement.length>0) {
                newChildren.push(<Text key='icon'>{React.cloneElement(iconElement[0], this.getIconProps(iconElement[0]))}</Text>);

                if (childrenArray[0].type==undefined) {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
                } else {
                    newChildren.push(<Text key='label' style={this.getTextStyle()}>{(Platform.OS==='ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
                }
            }
            else {
                return <Text style={this.getTextStyle()}>{(Platform.OS==='ios') ? this.props.children : this.props.children.toUpperCase()}</Text>
            }

            return newChildren;

        }

        else
            return React.cloneElement(this.props.children);

    }

    render() {
        return(
            <TouchableOpacity {...this.prepareRootProps()} >
                {this.renderChildren()}
            </TouchableOpacity>
        );
    }
}
