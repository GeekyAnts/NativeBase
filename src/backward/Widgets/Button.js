/* @flow */


import React, { Component } from 'react';
import { TouchableOpacity, Platform } from 'react-native';
import { connectStyle } from '@shoutem/theme';
import _ from 'lodash';
import variables from '../../theme/variables';
import { Icon } from './Icon';
import mapPropsToStyleNames from '../../Utils/mapPropsToStyleNames';
import computeProps from '../../Utils/computeProps';
import { Text } from './Text';


class Button extends Component {


  getInitialStyle() {
    return {
      borderedBtn: {
        borderWidth: (this.props.bordered) ? 1 : undefined,
        borderRadius: (this.props.rounded && this.props.bordered) ? variables.borderRadiusLarge : 2,
      },
    };
  }

  iconPresent() {
    let iconComponentPresent = false;
    React.Children.forEach(this.props.children, (child) => {
      if (child.type === Icon) { iconComponentPresent = true; }
    });
    return iconComponentPresent;
  }

  prepareRootProps() {
    const defaultProps = {
      style: this.getInitialStyle().borderedBtn,
    };

    return computeProps(this.props, defaultProps);
  }

  renderChildren() {
    if (typeof this.props.children === 'string') {
      return <Text style={this.props.textStyle}>{(Platform.OS === 'ios' || !this.props.capitalize) ? this.props.children : this.props.children.toUpperCase()}</Text>;
    } else if (this.props.children.type === Icon) {
      return React.cloneElement(this.props.children);
    } else if (Array.isArray(this.props.children)) {
      const newChildren = [];

      const childrenArray = React.Children.toArray(this.props.children);

      let iconElement = [];
      iconElement = _.remove(childrenArray, (item) => {
        if (item.type === Icon) {
          return true;
        }
        return null;
      });

      if (this.props.iconRight) {
        if (childrenArray[0].type === undefined) {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
        } else {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
        }

        newChildren.push(React.cloneElement(iconElement[0]));
      } else if (this.props.iconLeft || iconElement.length > 0) {
        newChildren.push(React.cloneElement(iconElement[0]));

        if (childrenArray[0].type === undefined) {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0] : childrenArray[0].toUpperCase()}</Text>);
        } else {
          newChildren.push(<Text key="label">{(Platform.OS === 'ios' || !this.props.capitalize) ? childrenArray[0].props.children : childrenArray[0].props.children.toUpperCase()}</Text>);
        }
      } else {
        return <Text>{this.props.children}</Text>;
      }

      return newChildren;
    }
    return React.cloneElement(this.props.children);
  }

  render() {
    return (
      <TouchableOpacity
        ref={(c) => { this._root = c; }}
        {...this.prepareRootProps()} activeOpacity={0.5}
      >
        {this.renderChildren()}
      </TouchableOpacity>
    );
  }
}


Button.propTypes = {
  ...TouchableOpacity.propTypes,
  style: React.PropTypes.object,
  textStyle: React.PropTypes.object,
  block: React.PropTypes.bool,
  primary: React.PropTypes.bool,
  transparent: React.PropTypes.bool,
  success: React.PropTypes.bool,
  danger: React.PropTypes.bool,
  warning: React.PropTypes.bool,
  info: React.PropTypes.bool,
  bordered: React.PropTypes.bool,
  disabled: React.PropTypes.bool,
  rounded: React.PropTypes.bool,
  large: React.PropTypes.bool,
  small: React.PropTypes.bool,
  inputButton: React.PropTypes.bool,
  tabButton: React.PropTypes.bool,
};


const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button);
export {
  StyledButton as Button,
};
