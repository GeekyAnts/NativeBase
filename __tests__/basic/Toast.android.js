import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Content } from "./../../src/basic/Content";
import { Button } from "./../../src/basic/Button";
import { Toast } from "./../../src/basic/Toast";
import { Text } from "./../../src/basic/Text";

// Note: test renderer must be required after react-native.
const cover = require("../assets/drawer-cover.png");

jest.mock("Platform", () => {
  const Platform = require.requireActual("Platform");
  Platform.OS = "android";
  return Platform;
});

it("renders default Toast", () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          Toast.show({
            text: "Wrong password!",
            buttonText: "Okay"
          })
        }
      >
        <Text>Toast</Text>
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Toast button style", () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          Toast.show({
            text: "Wrong password!",
            buttonText: "Okay",
            buttonTextStyle: { color: "#008000" },
            buttonStyle: { backgroundColor: "#5cb85c" }
          })
        }
      >
        <Text>Toast</Text>
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Toast with duration", () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          Toast.show({
            text: "Wrong password!",
            buttonText: "Okay",
            duration: 3000
          })
        }
      >
        <Text>Toast</Text>
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Toast position", () => {
  const tree = renderer
    .create(
      <Content padder>
        <Button
          onPress={() =>
            Toast.show({
              text: "Wrong password!",
              buttonText: "Okay",
              position: "top"
            })
          }
        >
          <Text>Top Toast</Text>
        </Button>
        <Button
          onPress={() =>
            Toast.show({
              text: "Wrong password!",
              buttonText: "Okay",
              position: "bottom"
            })
          }
        >
          <Text>Bottom Toast</Text>
        </Button>
      </Content>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Toast Text style", () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          Toast.show({
            text: "Wrong password!",
            textStyle: { color: "yellow" },
            buttonText: "Okay"
          })
        }
      >
        <Text>Toast</Text>
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Toast Type", () => {
  const tree = renderer
    .create(
      <Content padder>
        <Button
          onPress={() =>
            Toast.show({
              text: "Wrong password!",
              buttonText: "Okay"
            })
          }
        >
          <Text>Default Toast</Text>
        </Button>
        <Button
          success
          onPress={() =>
            Toast.show({
              text: "Wrong password!",
              buttonText: "Okay",
              type: "success"
            })
          }
        >
          <Text>Success Toast</Text>
        </Button>
        <Button
          warning
          onPress={() =>
            Toast.show({
              text: "Wrong password!",
              buttonText: "Okay",
              type: "warning"
            })
          }
        >
          <Text>Warning Toast</Text>
        </Button>
        <Button
          danger
          onPress={() =>
            Toast.show({
              text: "Wrong password!",
              buttonText: "Okay",
              type: "danger"
            })
          }
        >
          <Text>Danger Toast</Text>
        </Button>
      </Content>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
