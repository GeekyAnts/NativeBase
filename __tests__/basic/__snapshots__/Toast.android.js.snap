// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Toast Text style 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      Object {
        "borderRadius": 2,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }
  >
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;

exports[`renders Toast Type 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": 10,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  padder={true}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={Object {}}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          Object {
            "borderRadius": 2,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20,
          }
        }
      >
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          Object {
            "borderRadius": 2,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20,
          }
        }
      >
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          Object {
            "borderRadius": 2,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20,
          }
        }
      >
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          Object {
            "borderRadius": 2,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20,
          }
        }
      >
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders Toast button style 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      Object {
        "borderRadius": 2,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }
  >
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;

exports[`renders Toast position 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": 10,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  padder={true}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={Object {}}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          Object {
            "borderRadius": 2,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20,
          }
        }
      >
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          Object {
            "borderRadius": 2,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20,
          }
        }
      >
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders Toast with duration 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      Object {
        "borderRadius": 2,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }
  >
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;

exports[`renders default Toast 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      Object {
        "borderRadius": 2,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }
  >
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;
